rosidl_generator_cpp/resource/msg__builder.hpp.em

```diff
@@ -0,0 +1,85 @@
+@# Included from rosidl_generator_cpp/resource/idl__builder.hpp.em
+@{
+message_typename = '::'.join(message.structure.namespaced_type.namespaced_name())
+}@
+
+@[for ns in message.structure.namespaced_type.namespaces]@
+namespace @(ns)
+{
+
+@[end for]@
+@[if not message.structure.members]@
+::@(message_typename) build_@(message.structure.namespaced_type.name)()
+{
+return ::@(message_typename)(rosidl_generator_cpp::MessageInitialization::ZERO);
+}
+@[else]@
+namespace builder
+{
+
+@[  for index in range(len(message.structure.members) - 1, -1, -1)]@
+@{
+field_name = message.structure.members[index].name
+if index < len(message.structure.members) - 1:
+  next_field_name = message.structure.members[index + 1].name
+}@
+class Init_@(message.structure.namespaced_type.name)_@(field_name)
```

Comment:
I'd think that builders should be templated by allocator type just like interface structs are (see [here](https://github.com/ros2/rosidl/blob/4d837670d5758a3d45b8c86f3d867f44edd1c221/rosidl_generator_cpp/resource/msg__struct.hpp.em#L88-L89)).

Reply:
At the risk of beating a dead horse, we would get this trivially if we used the earlier static member function approach.

Reply:
Since none of the ROS 2 API works at the moment with custom allocators I would rather not make this more complicated and just leave custom allocators out of this.

Reply:
Hmm I do see `rclcpp::Publisher` propagating the given `AllocatorT` accordingly. I don't see any `MessageT` allocator being used though. So I'm fine with not doing anything else on this PR, but a follow-up issue to track the missing (or half-implemented or to be deprecated) feature would be nice.